datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -354,29 +353,75 @@ func (f *ChainDataFetcher) updateCheckpoint(num int64) error {
 	return nil
 }
 
-func (f *ChainDataFetcher) updateGauge(insert func(chainEvent blockchain.ChainEvent), gauge metrics.Gauge) func(chainEvent blockchain.ChainEvent) {
-	return func(chainEvent blockchain.ChainEvent) {
+func getInsertionTimeGauge(reqType cfTypes.RequestType) (metrics.Gauge, error) {
+	switch reqType {
+	case cfTypes.RequestTypeTransaction:
+		return txsInsertionTimeGauge, nil
+	case cfTypes.RequestTypeTokenTransfer:
+		return tokenTransfersInsertionTimeGauge, nil
+	case cfTypes.RequestTypeContract:
+		return contractsInsertionTimeGauge, nil
+	case cfTypes.RequestTypeTrace:
+		return tracesInsertionTimeGauge, nil
+	default:
+		return nil, fmt.Errorf("the request type is not supported. type: %v", reqType)
```

Comment:
If you leave a warning log and return an empty gauge, you don't need error handling logic at updateInsertionTimeGauge.

Reply:
Agreed. Reflected on a9b659884694577dba20dad99f4c46673df1aab3.